src/trie/merkle_tree.rs

```diff
@@ -91,23 +253,36 @@ pub struct MerkleTree<H: StarkHash, DB: BonsaiDatabase, ID: Id> {
 }
 
 #[derive(Debug, PartialEq, Eq)]
-enum InsertOrRemove<T> {
+pub(crate) enum InsertOrRemove<T> {
     Insert(T),
     Remove,
 }
 
-impl<H: StarkHash, DB: BonsaiDatabase, ID: Id> MerkleTree<H, DB, ID> {
+/// Note to developers : Must be used to build a key for the database.
+// Allow because needed for no-std
+#[allow(clippy::ptr_arg)]
+pub fn build_db_key(identifier: &Vec<u8>, key: &[u8]) -> Vec<u8> {
+    let mut db_key = identifier.clone();
+    db_key.extend_from_slice(key);
+    db_key
+}
```

Comment:
Can the key created by the concatenation of the storagekey and the identifier be it's own type?
So that it is enforced at the compiler level that the concatenation has been done and we are not just returning the storage key, which is also a `Vec<u8>`.
Otherwise future developer can easily write code that contains this error

Ideally we could create wrapper type for both the identifier and the trie key, and combine them to create a DatabaseKey